velox/connectors/hive/storage_adapters/s3fs/tests/S3FileSystemTest.cpp

```diff
@@ -113,7 +112,17 @@ TEST_F(S3FileSystemTest, missingFile) {
   auto hiveConfig = minioServer_->hiveConfig();
   filesystems::S3FileSystem s3fs(hiveConfig);
   s3fs.initializeClient();
-  EXPECT_THROW(s3fs.openFileForRead(s3File), VeloxException);
+  try {
+    s3fs.openFileForRead(s3File);
+    FAIL() << "Expected VeloxException";
+  } catch (VeloxException const& err) {
+    EXPECT_EQ(
+        err.message(),
+        std::string(
+            "Failed to initialize S3 file with bucket 'data1' and key 'i-do-not-exist.txt' due to 16:No response body."));
+  } catch (...) {
+    FAIL() << "Expected VeloxException";
```

Comment:
I'd remove this catch block for better error messages. As it is now, the error message will not contain the exception that was thrown.